Update analytics-solutions-teradata.md (#1228)
MicrosoftDocs/cloud-adoption-framework: docs/migrate/azure-best-practices/analytics/analytics-solutions-teradata.md

    `SELECT * FROM CUSTOMER WHERE (POSTCODE LIKE 'CV1%') OR (POSTCODE LIKE 'CV2%') OR (POSTCODE LIKE 'CV3%') ;`

- Depending on system settings, character comparisons in Teradata might not be case-specific by default. In Azure Synapse, these comparisons are always case-specific.

> - `Recursive views` are available in Teradata but not available in Azure Synapse

Reviewer: Please do expand the `CREATE RECURSIVE VIEW` topic here like the DML items above.  List the TD version, then a workaround (if any) for AS.

Author: There is no clear workaround for recursive views. This has to be handled on a case by case basis.

> - `Multi-column IN/NOT IN` clauses are supported in Teradata but not available in Azure Synapse

Reviewer: Please do expand the multi-column `IN`/`NOT IN` topic here like the DML items above.  List the TD version, then a workaround (if any) for AS.

- Teradata performs auto-truncation of data during insertion. For example, you can insert a VARCHAR(100) field into a VARCHAR(50) field. Although only the first 50 characters would be inserted, Teradata will not complain or warn about it.

> - Teradata can group by the ordinality of the column in the select clause. For example, `select employee, dept_id from emp_dept group by 1`. In Azure Synapse, this query needs to be written by replacing the ordinal position by the column name.

Reviewer: Please break this `group by` example out into the same format as the DML examples above.

Author: Example added

> - Teradata can use a column alias in the same query but Azue Synapse cannot. For example, you can write a query like `select employee as emp_name, dept_id where emp_name = 'Steve'`. In Azure Synapse, the column name would not be recognized.

Reviewer: Please break out the column alias example into the same format as the DML examples above.

Author: Example added

- Depending on system settings, character comparisons in Teradata might not be case-specific by default. In Azure Synapse, these comparisons are always case-specific.

> - Recursive views are available in Teradata but not available in Azure Synapse. There is no direct conversion of a recursive view in Azure Synapse. This has to be handed in a similar fashion of how you would flatten hierarchical data.

Reviewer 2: ```suggestion
- Recursive views are available in Teradata but are not available in Azure Synapse. There is no direct conversion of a recursive view in Azure Synapse. This has to be handled in a similar fashion to how you would flatten hierarchical data.
```

- Multi-column `IN` and `NOT IN` clauses are supported in Teradata but not available in Azure Synapse.
  
  For example:
  
  ```sql
  SELECT employee, dept_id 
  FROM emp_dept 
  WHERE (employee, dept_id) IN 
  (
    SELECT employee, dept_id
    FROM Ext_emp_dept
  )
  ```

  The converted query in Azure Synapse is:

  ```sql
  SELECT  employee, dept_id 
  FROM emp_dept ed
  WHERE EXISTS (
    SELECT 1
    FROM Ext_emp_dept eed
    WHERE ed.employee = edd.employee AND ed.dept_id = edd.dept_id
  )
  ```

> - Teradata performs auto-truncation of data during insertion. For example, you can insert a `VARCHAR(100)` field into a `VARCHAR(50)` field. Although only the first 50 characters would be inserted, Teradata will not complain or warn about it.

Reviewer 2: ```suggestion
- Teradata performs auto-truncation of data during insertion. For example, you can insert a `VARCHAR(100)` field into a `VARCHAR(50)` field. Although only the first 50 characters would be inserted, Teradata won't complain or warn about it.
```

- Teradata can group by the ordinality of the column in the select clause.
  
  For example:
  
  ```sql
  SELECT employee, dept_id FROM emp_dept GROUP BY 1, 2
  ```

  The converted query in Azure Synapse is written by replacing the ordinal position by the column name.

  ```sql
  SELECT employee, dept_id FROM emp_dept GROUP BY employee, dept_id
  ```

> - Teradata can use a column alias in the same query but Azue Synapse cannot.

Reviewer 2: ```suggestion
- Teradata can use a column alias in the same query but Azue Synapse can't.
```

  For example:
  
  ```sql
  SELECT employee as emp_name, dept_id FROM emp_dept WHERE emp_name = "Steve"
  ```
  

>   In Azure Synapse, the column name would not be recognized and you have to write the query as:

Reviewer 2: ```suggestion
  In Azure Synapse, the column name wouldn't be recognized, and you have to write the query as:
```
